docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+│   ├───System.Net.Sockets
+│   ├───System.Numerics.Vectors
+│   ├───System.Runtime
+│   ├───System.Runtime.Extensions
+│   ├───System.Runtime.Numerics
+│   ├───System.Runtime.Serialization.Formatters
+│   ├───System.Security.Cryptography
+│   ├───System.Security.Cryptography.Primitives
+│   ├───System.Text.Encoding
+│   ├───System.Text.RegularExpressions
+│   ├───System.Threading
+│   ├───System.Threading.Channels
+│   ├───System.Threading.Tasks
+│   ├───System.Threading.Tasks.Extensions
+│   ├───System.Threading.ThreadPool
+│   ├───System.Threading.Timers
+│   └───System.Xml.XmlDocument
+```
+
+During the port from xunit-performance to BenchmarkDotNet the namespaces, type and methods names were not changed. The exception to this rule are all `System.Collections` ([#92](https://github.com/dotnet/performance/pull/92)) and `Span<T>` ([#94](https://github.com/dotnet/performance/pull/94)) benchmarks which got rewritten to utilize the full capabilities of BenchmarkDotNet.
+
+Please remember that you can  filter the benchmarks using a glob pattern applied to namespace.typeName.methodName ([read more](./benchmarkdotnet.md#Filtering-the-Benchmarks)):
+
+```cmd
+dotnet run -f netcoreapp3.0 --filter System.Memory*
+```
+
+Moreover, every CoreFX benchmark belongs to a [CoreFX category](../src/benchmarks/micro/README.md#Categories)
+
+### CoreFX Prerequisites
+
+In order to run the benchmarks against local CoreFX build you need to build the CoreFX repository in **Release**:
+
+```cmd
+C:\Projects\corefx> build -Release
+```
+
+**The most important build artifact for us is CoreRun**. CoreRun is a simple host that does NOT take any dependency on NuGet. BenchmarkDotNet generates some boilerplate code, builds it using dotnet cli and tells CoreRun.exe to run the benchmarks from the auto-generated library. CoreRun runs the benchmarks using the libraries that are placed in its folder. When a benchmarked code has a dependency to `System.ABC.dll` version 4.5 and CoreRun has `System.ABC.dll` version 4.5.1 in its folder, then CoreRun is going to load and use `System.ABC.dll` version 4.5.1. **This is why having a single clone of .NET Performance repository allows you to run benchmarks against private builds of CoreCLR/FX from many different locations.**
+
+Every time you want to run the benchmarks against local build of CoreFX you need to provide the path to CoreRun:
+
+```cmd
+dotnet run -f netcoreapp3.0 --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" --filter $someFilter 
+```
+
+**Note:** BenchmarkDotNet expects a path to `CoreRun.exe` file (`corerun` on Unix), not to `Core_Root` folder.
+
+Once you rebuild the part of CoreFX you are working on, the appropriate `.dll` gets updated and the next time you run the benchmarks CoreRun is going to load the updated library.
+
+```cmd
+C:\Projects\corefx\src\System.Text.RegularExpressions\src> dotnet msbuild /p:ConfigurationGroup=Release
+```
+
+## Preventing Regressions
```

Comment:
@danmosemsft this is something new, I would like it to become a part of our perf culture (thinking of perf up-front and using tools to validate that)